src/github/client/key-locator.ts

```diff
@@ -0,0 +1,59 @@
+import { GitHubServerApp } from "~/src/models/github-server-app";
+import { resolve } from "path";
+import { existsSync, readFileSync } from "fs";
+import { envVars } from "~/src/config/env";
+import isBase64 from "is-base64";
+
+const PRIVATE_KEY_BEGIN = "-----BEGIN RSA PRIVATE KEY-----";
+const PRIVATE_KEY_END = "-----END RSA PRIVATE KEY-----";
+/**
+ * Look for a Github app's private key
+ */
+export const keyLocator = async (gitHubAppId?: number) => {
+	if (gitHubAppId) {
+		const gitHubServerApp = await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
+		if (gitHubServerApp) {
+			return await gitHubServerApp.decrypt("privateKey");
+		}
+	}
+	else {
+		// Fetch private key for github cloud
+		if (envVars.PRIVATE_KEY) {
+			let privateKey = envVars.PRIVATE_KEY;
+			if (isBase64(privateKey)) {
+				privateKey = Buffer.from(privateKey, "base64").toString();
+			}
+
+			if (privateKey.includes(PRIVATE_KEY_BEGIN) && privateKey.includes(PRIVATE_KEY_END)) {
+				// newlines are escaped
+				if (privateKey.indexOf("\\n") !== -1) {
+					privateKey = privateKey.replace(/\\n/g, "\n");
+				}
+				// newlines are missing
+				if (privateKey.indexOf("\n") === -1) {
+					privateKey = addNewlines(privateKey);
+				}
+				return privateKey;
```

Comment:
I'm fairly certain that we can just do `return envVars.PRIVATE_KEY`.  We shouldn't get a base64 key nor one that's been poorly formatted.  Having all of these "extra" steps to using the key is adding risks of side effects to how we handle the key.